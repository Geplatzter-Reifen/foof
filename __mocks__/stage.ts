type StageMockConstructorArgs = {
  title: string;
  isActive: boolean;
  startedAt: number;
  finishedAt: number | undefined;
  distance: number;
  avgSpeed: number;
  tour: { id: string; name: string };
  locations: [{ id: string; latitude: number; longitude: number }];
  collections: any;
};

export class MockStage {
  constructor({
    title = "Mock Stage",
    isActive = true,
    startedAt = new Date("2024-12-03 12:00").getTime(),
    finishedAt = undefined,
    distance = 42.195,
    avgSpeed = 10.5,
    tour = { id: "mock-tour-id", name: "Mock Tour" },
    locations = [{ id: "mock-location-id", latitude: 0, longitude: 0 }],
    collections = {},
  }: Partial<StageMockConstructorArgs> = {}) {
    this.title = title;
    this.isActive = isActive;
    this.startedAt = startedAt;
    this.finishedAt = finishedAt;
    this.distance = distance;
    this.avgSpeed = avgSpeed;
    this.tour = tour;
    this.collections = collections;
    this.locations = locations;
  }
  static table = "stages";

  static associations = {
    tours: { type: "belongs_to", key: "tour_id" },
    locations: { type: "has_many", foreignKey: "stage_id" },
  };

  // Properties
  title: string;
  isActive: boolean;
  startedAt: number;
  finishedAt?: number;
  distance: number;
  avgSpeed: number;
  tour: { id: string; name: string };
  locations: [{ id: string; latitude: number; longitude: number }];
  collections;
}
