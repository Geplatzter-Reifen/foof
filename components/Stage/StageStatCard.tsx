import React from "react";
import { DateFormat, formatDate, getDurationFormatted } from "@/utils/dateUtil";

import { Stage } from "@/database/model/model";

import { Button, Card, Icon, IconElement, Text } from "@ui-kitten/components";
import { ImageProps, StyleSheet, View } from "react-native";
import customStyles from "../../constants/styles";
import { withObservables } from "@nozbe/watermelondb/react";
import { shareStage } from "@/services/sharingService";

const ShareIcon = (props?: Partial<ImageProps>): IconElement => (
  <Icon
    {...props}
    name="share-nodes"
    style={[props?.style, { height: 20, width: "auto" }]}
  />
);

function StageStatCard({ stage }: { stage: Stage }) {
  const startedAt: Date = new Date(stage.startedAt);
  let finishedAt: Date | undefined = stage.finishedAt
    ? new Date(stage.finishedAt)
    : undefined;

  const date: string = formatDate(startedAt, DateFormat.DATE);

  let duration: string | undefined = finishedAt
    ? getDurationFormatted(startedAt, finishedAt)
    : getDurationFormatted(startedAt, new Date(Date.now()));

  const distance: string = stage.distance.toFixed(1);

  const avgSpeed: string = stage.avgSpeed.toFixed(1);

  const Header = () => {
    return (
      <View style={styles.header}>
        <Text category="h5" style={styles.title}>
          Statistiken
        </Text>
        <Button
          status="primary"
          appearance="ghost"
          accessoryLeft={ShareIcon}
          onPress={() => shareStage(stage)}
        />
      </View>
    );
  };

  const StartEnd = () => {
    return <View></View>;
  };

  return (
    <Card
      style={{
        ...customStyles.basicCard,
        ...customStyles.basicShadow,
        ...styles.card,
      }}
      header={<Header />}
    >
      <Text appearance="hint">vom {date}</Text>
      <StartEnd />
    </Card>
  );
}

// Observe die reingegebene Prop "stage"und reagiere auf Ã¤nderungen
const enhance = withObservables(["stage"], ({ stage }) => ({ stage }));
export default enhance(StageStatCard);

const styles = StyleSheet.create({
  card: {
    flex: 1,
    marginBottom: 5,
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  title: {
    marginTop: 10,
    marginLeft: 15,
    marginBottom: 10,
  },
  stat: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
  },
  icon: {
    marginRight: 10,
  },
  statLabel: {
    fontSize: 17,
  },
  date: {
    marginTop: 5,
  },
});
