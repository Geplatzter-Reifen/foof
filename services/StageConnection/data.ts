import { MapPoint } from "@/utils/locationUtil";

export const flensburg: MapPoint = {
  latitude: 54.789356,
  longitude: 9.434199,
};
export const oberstdorf: MapPoint = {
  latitude: 47.408333325,
  longitude: 10.27931564,
};
export const hannover: MapPoint = {
  latitude: 52.3744779,
  longitude: 9.7385532,
};
export const wuerzburg: MapPoint = {
  latitude: 49.7933723,
  longitude: 9.9309779,
};
export const wiesbaden: MapPoint = {
  latitude: 52.520007,
  longitude: 13.404954,
};
