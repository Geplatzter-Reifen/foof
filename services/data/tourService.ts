import { Q } from "@nozbe/watermelondb";
import { database } from "@/model/createDatabase";
import { Tour } from "@/model/model";
import { getAllStagesByTourId } from "@/services/data/stageService";
import { getAllLocationsByStageId } from "@/services/data/locationService";

// READ

export const getAllToursQuery = database.get<Tour>("tours").query();
export const getAllTours = () => getAllToursQuery.fetch();

export const getTourByTourIdQuery = (tourId: string) => {
  return database.get<Tour>("tours").query(Q.where("id", tourId));
};
export const getTourByTourId = (tourId: string) => {
  return database.get<Tour>("tours").find(tourId);
};

export const getActiveTour = async (): Promise<Tour | null> => {
  const activeTours = await database
    .get<Tour>("tours")
    .query(Q.where("is_active", true), Q.take(1))
    .fetch();
  if (activeTours.length === 0) {
    return null;
  } else {
    return activeTours[0];
  }
};

// CREATE

export const createTour = async (
  title: string,
  startedAt?: number,
): Promise<Tour> => {
  return database.write(async () => {
    return database.get<Tour>("tours").create((tour) => {
      tour.title = title;
      tour.isActive = false;
      if (startedAt) {
        tour.startedAt = startedAt;
      }
    });
  });
};

// UPDATE

export const setTourActive = async (tourId: string) => {
  await database.write(async () => {
    const allTours = await database.get<Tour>("tours").query().fetch();
    for (const tour of allTours) {
      if (tour.isActive) {
        await tour.update(() => {
          tour.isActive = false;
        });
      }
    }
    const tour = await database.get<Tour>("tours").find(tourId);
    await tour.update(() => {
      tour.isActive = true;
    });
  });
};

export const setTourInactive = async (tourId: string) => {
  await database.write(async () => {
    const tour = await database.get<Tour>("tours").find(tourId);
    await tour.update(() => {
      tour.isActive = false;
    });
  });
};

export const updateTourNameById = async (
  tourId: string,
  newTourName: string,
) => {
  await database.write(async () => {
    const tour = await getTourByTourId(tourId);
    await tour.update(() => {
      tour.title = newTourName;
    });
  });
};

// DELETE

export const deleteTour = (tourId: string) => {
  void database.write(async () => {
    const tour = await getTourByTourId(tourId);
    const stages = await getAllStagesByTourId(tour.id);
    for (const stage of stages) {
      const locations = await getAllLocationsByStageId(stage.id);
      for (const location of locations) {
        await location.destroyPermanently();
      }
      await stage.destroyPermanently();
    }
    await tour.destroyPermanently();
  });
};

export const deleteAllTours = () => {
  void database.write(async () => {
    const tours = await getAllTours();
    for (const tour of tours) {
      const stages = await getAllStagesByTourId(tour.id);
      for (const stage of stages) {
        const locations = await getAllLocationsByStageId(stage.id);
        for (const location of locations) {
          await location.destroyPermanently();
        }
        await stage.destroyPermanently();
      }
      await tour.destroyPermanently();
    }
  });
};
