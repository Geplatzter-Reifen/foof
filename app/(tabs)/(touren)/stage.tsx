import {
  Input,
  Layout,
  Text,
  Icon,
  useTheme,
  ThemeType,
} from "@ui-kitten/components";
import { useLocalSearchParams, useNavigation, router } from "expo-router";
import React, { useCallback, useEffect, useMemo, useState } from "react";
import {
  deleteStage,
  getStageByStageId,
  setStageTitle,
} from "@/services/data/stageService";
import { getAllLocationsByStageId } from "@/services/data/locationService";
import { Location, Stage } from "@/database/model/model";
import { Alert, StyleSheet, TouchableOpacity, View } from "react-native";
import StageStatCard from "@/components/Stage/StageStatCard";
import StageMapView from "@/components/Stage/StageMapView";
import ConfirmDialog from "@/components/Dialog/ConfirmDialog";
import { tourIsFinished } from "@/services/StageConnection/stageConnection";
import { getActiveTour } from "@/services/data/tourService";

export default function StageScreen() {
  // Etappen-ID aus dem Pfad holen
  const { stageId } = useLocalSearchParams<{ stageId: string }>();
  // Lokale Variablen für die Etappe, Etappenlocations und Etappentitel
  const [stage, setStage] = useState<Stage>();
  const [stageLocations, setStageLocations] = useState<Location[]>([]);
  const [stageTitleVal, setStageTitleVal] = useState("Etappe");
  // Wird der Titel gerade bearbeitet?
  const [titleBeingChanged, setTitleBeingChanged] = useState(false);
  // Um den Header anzupassen, wenn der Etappentitel bearbeitet wird
  const navigation = useNavigation();
  // Wird der Löschen-Dialog angezeigt?
  const [isDeleteDialogVisible, setIsDeleteDialogVisible] = useState(false);

  const theme = useTheme();
  const styles = makeStyles(theme);

  // Vorbereitung: Etappe, Etappentitel und Locations setzen
  useEffect(() => {
    (async () => {
      const dbStage = await getStageByStageId(stageId);
      if (dbStage) {
        setStage(dbStage);
        setStageTitleVal(stage?.title ?? "Etappe");
      } else {
        router.back();
        Alert.alert("Stage does not exist");
      }
      const dbLocations = await getAllLocationsByStageId(stageId);
      if (dbLocations) {
        setStageLocations(dbLocations);
      }
    })();
  }, [stage, stageId]);

  // Button zum Löschen der Etappe & zurücknavigieren
  const DeleteButton = useCallback(() => {
    const handleConfirm = async () => {
      deleteStage(stageId).then(() => {
        getActiveTour().then(async (tour) => {
          if (tour != null) {
            tourIsFinished(tour).then((result) => {
              if (result) {
                Alert.alert("Tour beendet", "Herzlichen Glückwunsch!");
              }
            });
          }
        });
      });

      setIsDeleteDialogVisible(false);
      navigation.goBack();
    };

    return (
      <View style={styles.deleteButton}>
        <TouchableOpacity
          onPress={() => {
            setIsDeleteDialogVisible(true);
          }}
        >
          <Icon name="trash-can" style={styles.deleteIcon} />
        </TouchableOpacity>

        {/*Öffnet einen Dialog zum Bestätigen*/}
        <ConfirmDialog
          visible={isDeleteDialogVisible}
          title="Etappe löschen"
          message={`Möchtest du die Etappe \"${stage?.title}\" wirklich löschen?`}
          confirmString="Löschen"
          onConfirm={handleConfirm}
          onCancel={() => {
            setIsDeleteDialogVisible(false);
          }}
        />
      </View>
    );
  }, [
    isDeleteDialogVisible,
    navigation,
    stage?.title,
    stageId,
    styles.deleteButton,
    styles.deleteIcon,
  ]);

  // Input-Feld, in dem der Etappentitel eingegeben wird
  const titleInput = useMemo(
    () => (
      <Layout style={styles.headerInput}>
        <Input
          value={stageTitleVal}
          onChangeText={(nextValue) => setStageTitleVal(nextValue)}
          maxLength={20}
        />
      </Layout>
    ),
    [styles.headerInput, stageTitleVal],
  );

  // Edit/Confirm-Button, der den Bearbeitungsmodus (de)aktiviert und den neuen Titel in die DB Schreibt
  const changeTitleButton = useMemo(
    () => (
      <TouchableOpacity
        onPress={async () => {
          setTitleBeingChanged((prevState) => !prevState);
          if (titleBeingChanged && stage) {
            await setStageTitle(stage.id, stageTitleVal);
          }
        }}
      >
        {titleBeingChanged ? (
          <Icon name="check" style={styles.iconStyle} />
        ) : (
          <Icon name="edit" style={styles.iconStyle} />
        )}
      </TouchableOpacity>
    ),
    [styles.iconStyle, stage, stageTitleVal, titleBeingChanged],
  );

  // Header: zwischen Etappentitel und Eingabefeld switchen
  useEffect(() => {
    navigation.setOptions({
      headerTitle: () =>
        titleBeingChanged ? (
          titleInput
        ) : (
          <Text category="h4">{stageTitleVal}</Text>
        ), // Switch between input and title
      headerRight: () => (
        <View style={{ flexDirection: "row" }}>
          {changeTitleButton}
          <DeleteButton />
        </View>
      ), // Set the view on the right
      headerTitleAlign: "center", // Keep the title centered
    });
  }, [
    navigation,
    titleInput,
    changeTitleButton,
    titleBeingChanged,
    stageTitleVal,
    DeleteButton,
  ]);

  return (
    <Layout level="2" style={styles.layout}>
      {/*Kachel mit den Statistiken der Etappe*/}
      {stage && stageLocations.length !== 0 && (
        <StageStatCard stage={stage} locations={stageLocations} />
      )}
      {/*Falls keine / nur eine Location vorhanden ist: Hinweistext*/}
      {stage && stageLocations.length < 2 && (
        <Text style={{ textAlign: "center" }}>
          Diese Etappe ist leer. Bitte lösche sie.
        </Text>
      )}
      {/*Karte, auf der die gefahrene Etappe angezeigt wird*/}
      {stageLocations.length > 1 && (
        <View style={styles.mapContainer}>
          {stage && <StageMapView stage={stage} locations={stageLocations} />}
        </View>
      )}
    </Layout>
  );
}

const makeStyles = (theme: ThemeType) => {
  return StyleSheet.create({
    iconStyle: {
      height: 25,
      width: "auto",
      color: theme["color-primary-500"],
    },
    deleteButton: {
      marginLeft: 13,
    },
    deleteIcon: {
      height: 25,
      width: "auto",
      color: theme["color-basic-400"],
    },
    layout: {
      flex: 1,
      paddingVertical: 10,
      paddingHorizontal: 10,
    },
    headerInput: {
      justifyContent: "space-around",
      alignItems: "center",
    },
    mapContainer: {
      flex: 1,
      borderRadius: 8,
      overflow: "hidden", // Wichtig für abgerundete Ecken
      marginBottom: 5,
    },
  });
};
