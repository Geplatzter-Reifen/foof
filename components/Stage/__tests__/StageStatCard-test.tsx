import { render } from "@/test-utils/test-utils";
import { fetchPlaceName } from "@/services/geoService";
import StageStatCard from "@/components/Stage/StageStatCard";
import { createTour } from "@/services/data/tourService";
import {
  createStage,
  setStageAvgSpeed,
  setStageDistance,
} from "@/services/data/stageService";
import { waitFor } from "@testing-library/react-native";
import {
  createLocation,
  getAllLocationsByStageId,
} from "@/services/data/locationService";
import { MockStage } from "@/__mocks__/stage";

jest.mock("../../../services/geoService", () => ({
  fetchPlaceName: jest.fn(),
}));

describe("StageStatCard", () => {
  beforeEach(() => {
    jest
      .spyOn(global.Date, "now")
      .mockImplementation(() => new Date("2024-12-03 19:21").getTime());
  });

  it("does not render Start and End Point for Stage with less than two Locations", () => {
    const stage = new MockStage();

    // @ts-ignore - TS mag die MockStage nicht
    const view = render(<StageStatCard stage={stage} locations={[]} />);
    expect(view).toMatchSnapshot();
  });

  it("renders correctly", async () => {
    (fetchPlaceName as jest.Mock).mockResolvedValueOnce("Startpunkt");
    (fetchPlaceName as jest.Mock).mockResolvedValueOnce("Endpunkt");

    const tour = await createTour("Test Tour");
    const stage = await createStage(
      tour.id,
      "Basic Stage",
      new Date("2024-12-01T10:28").getTime(),
      new Date("2024-12-01T17:43").getTime(),
      false,
    );
    await setStageDistance(stage.id, 79.345);
    await setStageAvgSpeed(stage.id, 10.9379);
    await createLocation(
      stage.id,
      50.3,
      8.9,
      new Date("2024-12-01T10:28").getTime(),
    );
    await createLocation(
      stage.id,
      50.8,
      8.1,
      new Date("2024-12-01T17:43").getTime(),
    );
    const locations = await getAllLocationsByStageId(stage.id);

    const view = await waitFor(() =>
      render(<StageStatCard stage={stage} locations={locations} />),
    );
    expect(view).toMatchSnapshot();
    // korrektes Datum
    expect(view.getByText("vom 01.12.2024")).toBeTruthy();
    // korrekte Start- und Zielzeit
    expect(view.getByText("10:28")).toBeTruthy();
    expect(view.getByText("17:43")).toBeTruthy();
    // korrekter Start- und Zielort
    expect(view.getByText("Startpunkt")).toBeTruthy();
    expect(view.getByText("Endpunkt")).toBeTruthy();
    // korrekte Statistiken
    expect(view.getByText("79.3 km")).toBeTruthy();
    expect(view.getByText("7h 15m")).toBeTruthy();
    expect(view.getByText("10.9 km/h")).toBeTruthy();
  });
});
