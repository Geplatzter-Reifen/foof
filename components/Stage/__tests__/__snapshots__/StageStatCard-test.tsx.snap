// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageStatCard does not render Start and End Point for Stage with less than two Locations 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#7e7e7e",
      "borderRadius": 8,
      "borderWidth": 0,
      "elevation": 3,
      "marginBottom": 15,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": undefined,
        "height": 0,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "baseline",
          "flexDirection": "row",
        }
      }
    >
      <Text
        appearance="default"
        category="h5"
        status="basic"
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 22,
              "fontWeight": "800",
            },
            {
              "marginBottom": 10,
              "marginLeft": 15,
              "marginTop": 10,
            },
          ]
        }
      >
        Statistiken
      </Text>
      <Text
        appearance="hint"
        category="p1"
        status="basic"
        style={
          [
            {
              "color": "#494949",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
            },
            {
              "fontSize": 18,
              "marginLeft": 10,
            },
          ]
        }
      >
        vom 
        03.12.2024
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 40,
          "minWidth": 40,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 12,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    appearance="default"
    style={
      [
        {
          "backgroundColor": "#b8b8b8",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "flexShrink": 1,
        },
        {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      style={
        {
          "gap": 3,
          "justifyContent": "space-between",
          "marginTop": 15,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
        testID="container"
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text />
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "flexShrink": 1,
                "flexWrap": "wrap",
                "fontSize": 20,
                "marginLeft": 8,
              },
            ]
          }
        >
          42.2 km
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
        testID="container"
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text />
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "flexShrink": 1,
                "flexWrap": "wrap",
                "fontSize": 20,
                "marginLeft": 8,
              },
            ]
          }
        >
          7h 21m
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
        testID="container"
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text />
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "flexShrink": 1,
                "flexWrap": "wrap",
                "fontSize": 20,
                "marginLeft": 8,
              },
            ]
          }
        >
          5.7 km/h
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`StageStatCard renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#7e7e7e",
      "borderRadius": 8,
      "borderWidth": 0,
      "elevation": 3,
      "marginBottom": 15,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": undefined,
        "height": 0,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "baseline",
          "flexDirection": "row",
        }
      }
    >
      <Text
        appearance="default"
        category="h5"
        status="basic"
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 22,
              "fontWeight": "800",
            },
            {
              "marginBottom": 10,
              "marginLeft": 15,
              "marginTop": 10,
            },
          ]
        }
      >
        Statistiken
      </Text>
      <Text
        appearance="hint"
        category="p1"
        status="basic"
        style={
          [
            {
              "color": "#494949",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
            },
            {
              "fontSize": 18,
              "marginLeft": 10,
            },
          ]
        }
      >
        vom 
        01.12.2024
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 40,
          "minWidth": 40,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 12,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FF6E14",
                "fontSize": 20,
              },
              {
                "marginHorizontal": 10,
                "width": "auto",
              },
              {
                "fontFamily": "FontAwesome6Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    appearance="default"
    style={
      [
        {
          "backgroundColor": "#b8b8b8",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "flexShrink": 1,
        },
        {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          {
            "flex": 6,
            "gap": 7,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
          testID="container"
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 23,
                  },
                  {
                    "color": "#494949",
                    "width": 25,
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                {
                  "flexShrink": 1,
                  "flexWrap": "wrap",
                  "fontSize": 18,
                  "marginLeft": 8,
                },
              ]
            }
          >
            Startpunkt
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
          testID="container"
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 23,
                  },
                  {
                    "color": "#494949",
                    "width": 25,
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                {
                  "flexShrink": 1,
                  "flexWrap": "wrap",
                  "fontSize": 18,
                  "marginLeft": 8,
                },
              ]
            }
          >
            Endpunkt
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "gap": 7,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          appearance="hint"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#494949",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 17,
                "textAlign": "right",
              },
            ]
          }
        >
          10:28
        </Text>
        <Text
          appearance="hint"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#494949",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "fontSize": 17,
                "textAlign": "right",
              },
            ]
          }
        >
          17:43
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "gap": 3,
          "justifyContent": "space-between",
          "marginTop": 15,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
        testID="container"
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "flexShrink": 1,
                "flexWrap": "wrap",
                "fontSize": 20,
                "marginLeft": 8,
              },
            ]
          }
        >
          79.3 km
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
        testID="container"
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "flexShrink": 1,
                "flexWrap": "wrap",
                "fontSize": 20,
                "marginLeft": 8,
              },
            ]
          }
        >
          7h 15m
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
        testID="container"
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": undefined,
                  "fontSize": 22,
                },
                {
                  "color": "#FF6E14",
                  "width": "auto",
                },
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "flexShrink": 1,
                "flexWrap": "wrap",
                "fontSize": 20,
                "marginLeft": 8,
              },
            ]
          }
        >
          10.9 km/h
        </Text>
      </View>
    </View>
  </View>
</View>
`;
